### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by coolKeen** [hzwer/arXiv2020-RIFE#7]
Thanks you for your nice work!!!
Do you have any plan for releasing training code?

Thanks in advance!

**Comment by hzwer** [comment]
Yes. But I’m currently busy sorting out the test scripts, I’m not sure if I can get it done this week.

**Comment by coolKeen** [comment]
> Yes. The key is that my method also needs to generate optical flow labels, and I plan to package this part of the results and upload them to google drive. But I’m currently busy sorting out the test scripts, I’m not sure if I can get it done this week.

Does generate optical flow labels means calculating optimal flow  F2->1 and F2->3 for  triplet (I1,I2,I3) in vimeo90k  dataset? If so, I can help you generate them using optimal flow model like PWCNet or RAFT if you think that is ok.

**Comment by hzwer** [comment]
I have generated them. Please wait until this weekend, I think I can deal with all then.

**Comment by hzwer** [comment]
There were too many pending issues last week. Due to the workload, the priority of this issue was lowered.

**Comment by Doubiiu** [comment]
Hello, Zhewei. Since reproducibility is very important. I am trying to reproduce your results by training your model from scratch. Hope you can put this work in a higher priority. Thanks very much for releasing the code!

**Comment by hzwer** [comment]
> Hello, Zhewei. Since reproducibility is very important. I am trying to reproduce your results by training your model from scratch. Hope you can put this work in a higher priority. Thanks very much for releasing the code!

I know, but I'm not a full time researcher or student. And the paper has not been published, so my support may be limited.

**Comment by hzwer** [comment]
链接:https://pan.baidu.com/s/1IV3XbFpNyxko0CXd5ru_fw  密码:zlp7

**Comment by hzwer** [comment]
Since the processed optical flow labels reach dozens of Gs, I plan to provide a small data set for testing the entire training pipeline. Friends who are interested in complete reproduction can use Liteflownet to generate labels.

**Comment by xxh96** [comment]
大佬您好，您用的是RAFT生成的光流 flow_gt吗？我在论文中看到有参考该文章，是用的demo模型还是自己重新训练了新的模型？

**Comment by hzwer** [comment]
> 大佬您好，您用的是RAFT生成的光流 flow_gt吗？我在论文中看到有参考该文章，是用的demo模型还是自己重新训练了新的模型？

我是用 liteflownet 的 demo 模型（如果稍微 finetune 一下会涨 0.1~0.2)，用 RAFT 会掉点 0.2

**Comment by xxh96** [comment]
在RIFE.py中228行，计算L_dis的  loss_cons += self.epe(flow_list[i], flow_gt[:, :2], 1)，flow_list[i] 个多维的，torch.Size([36, 2, H, H])，flow_gt应该如何对应呢，一般来说生成的光流size为（[1,2,H,H]）,我是最近开始研究插帧的，也不知道对不对。

**Comment by hzwer** [comment]
> 在RIFE.py中228行，计算L_dis的 loss_cons += self.epe(flow_list[i], flow_gt[:, :2], 1)，flow_list[i] 个多维的，torch.Size([36, 2, H, H])，flow_gt应该如何对应呢，一般来说生成的光流size为（[1,2,H,H]）,我是最近开始研究插帧的，也不知道对不对。

36 是 batchsize 吧，对于每对图片都要生成光流呀

**Comment by xxh96** [comment]
对的，我注意到了。我的错。

**Comment by xxh96** [comment]
您用的是v3版本吗？

**Comment by hzwer** [comment]
> 您用的是v3版本吗？

我用的是 v1, https://github.com/sniklaus/pytorch-liteflownet，我没有去研究那个 caffe。

**Comment by xxh96** [comment]
您有试过，pyflow吗？我看STARnet中，https://github.com/alterzero/STARnet用了相似的方法，
![image](https://user-images.githubusercontent.com/36263119/100847472-d78f8280-34ba-11eb-8300-a147587a0e0c.png)

**Comment by hzwer** [comment]
你说的 pyflow 是什么？可以让 flow t->0 = 插值 (flow t->0, - flow t->1) 来降 noise，但是训练结果差不多

**Comment by xxh96** [comment]
pyflow是STAR得到真实flow的方法，他是训练时实时生成的光流，没有另外生成标签。

**Comment by xxh96** [comment]
非常感谢你的回复。

**Comment by hzwer** [comment]
> pyflow是STAR得到真实flow的方法，他是训练时实时生成的光流，没有另外生成标签。

我也试过训练同时生成光流，会使训练时间 2x，感觉很亏。

**Comment by xxh96** [comment]
是的，pyflow好像是用c++实现的。实时性应该很快。

**Comment by hzwer** [comment]
We have added code and README. If there are other training issues, please create a new issue.
@coolKeen @Doubiiu
